server/controllers/clubProfileController.js

```diff
@@ -82,23 +84,37 @@ module.exports = {
     },
     create: function(req, res) {
         // TODO; req.body contains profile information
-        clubProfile.create(req.body)
-                .then(newclubProfile => res.json(newclubProfile))
+        clubProfile.create(req.body) 
+                .then(newclubProfile => {
+                  //set new profile's acc ID to the given accID
+                  newclubProfile.clubAccountId = req.params.accountId;
+                  //get the account assosiated with given accID and set its profile Id to the new club profile Id
+                  var newClubAccount = getById(req.params.accountId,res);
```

Comment:
here, if you pass res (our output stream to feed back to frontend) into this function, it will just directly return to frontend within the `getById` function instead of executing the rest of the code here! I think the easiest way to solve this problem is just to replicate the `getById` code here?

Reply:
Also, note that `getById` on its own accepts `req, res` as arguments, and try to access `req.params.id` for its own use. Therefore, we can't just pass in `req.params.accountId` or even `req` as the first argument into `getById`. The former would have an invalid format, and the latter doesn't have the required field (only has field `accountId`, not `id`).